### ISSUE TITLE
Improvements to K-Elbow Visualizer

### ISSUE THREAD

**Issue by bbengfort** [DistrictDataLabs/yellowbrick#515]
The following tasks from #91 are left over to finalize the implementation of the `KElbowVisualizer ` and improve it. 

- [ ] Add inertia metric as seen in [sklearn KMeans Equivalent of Elbow Method](https://stackoverflow.com/questions/41540751/sklearn-kmeans-equivalent-of-elbow-method)
- [ ] Improve documentation with a description of how to use K-Elbow to select `k` 

Stretch goals:

- [ ] Find example dataset with clear elbow curve demonstration to use in documentation 
- [ ] Parallelize with joblib and `n_jobs` argument  

The stretch goals can be converted to their own issues if not completed in this feature request.

**Comment by rebeccabilbro** [comment]
Hi there @lacanlale - thanks for checking out Scikit-Yellowbrick!
Yes, for enhancing the documentation of the `KElbowVisualizer`, the updates should go in the .rst files that generate the Sphinx docs (in this case, `yellowbrick/docs/api/cluster/elbow.rst`). As you can see, the [current version](http://www.scikit-yb.org/en/develop/api/cluster/elbow.html#elbow-method) of the elbow visualizer doc page is pretty thin on description of the use cases and work flow for doing k-selection, so there's a lot of potential for fleshing out the page with some more instructive paragraphs! You may also want to check out the section on k-elbow in the [examples.ipynb document](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/examples/examples.ipynb), which might have some useful content that could be repurposed here.

**Comment by bbengfort** [comment]
@lacanlale sorry for the confusion, I'm happy to clarify with the story so far. Right now our elbow method uses 3 metrics, distortion score, calinski-harbaz, and silhouette. Of these, [distortion scores are most commonly used to create elbows](https://pythonprogramminglanguage.com/kmeans-elbow-method/) (in fact, there is some question about whether or not the other two metrics create elbows at all, but they do create sharp peaks, which are easy to visually identify). 

scikit-learn's [KMeans implementation uses inertia](http://scikit-learn.org/stable/modules/clustering.html#k-means) (a distortion-like metric) to optimize its clusters, and stores those values in the estimator as `inertia_`, meaning that [other](https://www.linkedin.com/pulse/finding-optimal-number-clusters-k-means-through-elbow-asanka-perera/) [implementations](https://avidml.wordpress.com/2016/10/29/easily-understand-k-means-clustering/) of the elbow visualizer simply use this property. 

Here's the catch, though - this is only available on `KMeans`. As a result, we simply implemented our [own distortion score metric](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/cluster/elbow.py#L45). However, inertia is different enough that it might be useful and it improves the performance of the visualizer since an extra computation is not needed so we'd like to add it in as a (non-default) metric:

1. Add this metric to the list of valid metrics 
2. Compute this metric by simply grabbing it off of each KMeans estimator from `inertia_` 
3. Raise an exception if the clustering estimator does not have the `inertia_` property 
4. Add documentation to describe when inertia should be used over distortion and what the difference is.  

Hopefully that helps and provides some additional background. Let me know if you have any additional questions!

**Comment by Kautumn06** [comment]
Hi @lacanlale, thank you for contributing to Yellowbrick! KMeans tries to separate the samples into `n` groups of equal variance, minimizing inertia which is also known as within-cluster sum-of-squares (WCSS). I highly recommend [scikit-learn's documentation](http://scikit-learn.org/stable/modules/clustering.html#k-means) and I'm also tagging @mattharrison, who had originally [commented](https://github.com/DistrictDataLabs/yellowbrick/issues/91#issuecomment-372855036) on #91 about the possibility of adding `inertia_` as a metric to the `KElbowVisualizer`, so he might be willing to chime in about how he uses it in his own work or explains it to his students.

**Comment by lacanlale** [comment]
Thanks for the responses (both of you!) This clarifies a lot, and I'll definitely be looking into it!

**Comment by vivienneinus** [comment]
Hi,
I found the dataset better for Elbow Method regarding the stretch goal - Find example dataset with clear elbow curve demonstration to use in documentation. The dataset is created with make_blobs of 500 points with 8 features and 4 centroids.
![images_calinski_harabaz](https://user-images.githubusercontent.com/46324391/57234377-7063d080-6fee-11e9-8310-4b532ee497d8.png)

**Comment by GrahamStein** [comment]
Unless anyone is close to completion, I'm at Pycon 2019 Sprints and I'm going to take a look into adding in   @##inertia_.

**Comment by lwgray** [comment]
@GrahamStein were you still interested in tackling this issue?

**Comment by GrahamStein** [comment]
I have it done, just need to clean it up a bit. Sadly I'm booked out
working on life/house stuff for the remainder of the week.
